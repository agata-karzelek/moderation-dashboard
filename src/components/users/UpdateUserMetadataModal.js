import React, { useState, useEffect } from "react";
import { DialogTitle, Dialog, Typography, IconButton } from "@material-ui/core";
import CloseIcon from "@material-ui/icons/Close";
import { useStyles } from "../../style/createModerationModal";
import { editUserMetadata } from "../../services/pubnub";
import SnackBar from "../core/SnackBar";
import { showError } from "../../utils/helpers";
import { useFormik } from "formik";
import AddUserForm from "./AddUserForm";
import { userValidationSchema } from "./AddUserMetadataModal";

export default function UpdateUserMetadataModal(props) {
  const { pubnubObject, open, setOpen, data, isUpdated } = props;
  const [userName, setUserName] = useState("");
  const [email, setEmail] = useState(null);
  const [UUID, setUUID] = useState("");
  const [profileUrl, setProfileUrl] = useState(null);
  const [updateUserAlert, setUpdateUserAlert] = useState({
    success: { status: false, msg: "" },
    error: { status: false, msg: "" },
    loading: false,
  });
  const classes = useStyles();

  useEffect(() => {
    setUserName(data.name);
    setEmail(data.email);
    setUUID(data.id);
    setProfileUrl(data.profileUrl);
  }, [data]);

  const handleClose = () => {
    setOpen(false);
  };

  const formik = useFormik({
    initialValues: {
      email: !email ? "" : email,
      name: userName,
      user_id: UUID,
      profile_url: !profileUrl ? "" : profileUrl,
    },
    enableReinitialize: true,
    validationSchema: userValidationSchema,
    onSubmit: (values) => {
      updateUserMetadata(values);
    },
  });

  const updateUserMetadata = (values) => {
    setUpdateUserAlert({
      ...updateUserAlert,
      success: { status: false, msg: "" },
      error: { status: false, msg: "" },
      loading: true,
    });
    (async () => {
      try {
        const userResponse = await editUserMetadata(
          pubnubObject,
          values.name,
          values.email,
          values.user_id,
          values.profile_url
        );
        handleClose();
        setUpdateUserAlert({
          ...updateUserAlert,
          success: { status: true, msg: "User updated successfully" },
          error: { status: false, msg: "" },
          loading: false,
        });
        isUpdated(userResponse);
      } catch (e) {
        handleClose();
        setUpdateUserAlert({
          ...updateUserAlert,
          success: { status: false, msg: "" },
          error: { status: true, msg: showError(e.status.errorData) },
          loading: false,
        });
      }
    })();
  };

  return (
    <div>
      <Dialog
        fullWidth={true}
        onClose={handleClose}
        aria-labelledby="customized-dialog-title"
        open={open}
        className={classes.modalLayout}
      >
        <DialogTitle id="customized-dialog-title" onClose={handleClose} disableTypography>
          <Typography variant="h6" className={classes.headingFont}>
            Update User Metadata
          </Typography>
          {open ? (
            <IconButton aria-label="close" onClick={handleClose} className={classes.closeButton}>
              <CloseIcon />
            </IconButton>
          ) : null}
        </DialogTitle>
        <AddUserForm
          formik={formik}
          userAlert={updateUserAlert}
          handleClose={handleClose}
          buttonTitle={"UPDATE"}
          disabled={true}
        />
      </Dialog>
      {updateUserAlert.error.status && <SnackBar msg={updateUserAlert.error.msg} status={"info"} />}
      {updateUserAlert.success.status && (
        <SnackBar msg={updateUserAlert.success.msg} status={"success"} />
      )}
    </div>
  );
}
