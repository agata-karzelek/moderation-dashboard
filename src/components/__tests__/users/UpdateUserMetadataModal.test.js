import { shallow } from "enzyme";
import UpdateUserMetadataModal from "../../users/UpdateUserMetadataModal";
describe("update User meta data Modal", () => {
  it("should render correctly with no props", () => {
    const component = shallow(<UpdateUserMetadataModal />);
    expect(component).toMatchSnapshot();
  });
});
