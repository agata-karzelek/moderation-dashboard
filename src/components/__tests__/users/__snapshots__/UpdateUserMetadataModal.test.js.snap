// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update User meta data Modal should render correctly with no props 1`] = `
<div>
  <WithStyles(ForwardRef(Dialog))
    aria-labelledby="customized-dialog-title"
    className="makeStyles-modalLayout-8"
    fullWidth={true}
    onClose={[Function]}
  >
    <WithStyles(ForwardRef(DialogTitle))
      disableTypography={true}
      id="customized-dialog-title"
      onClose={[Function]}
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-headingFont-13"
        variant="h6"
      >
        Update User Metadata
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(DialogTitle))>
    <AddUserForm
      buttonTitle="UPDATE"
      disabled={true}
      formik={
        Object {
          "dirty": false,
          "errors": Object {},
          "getFieldHelpers": [Function],
          "getFieldMeta": [Function],
          "getFieldProps": [Function],
          "handleBlur": [Function],
          "handleChange": [Function],
          "handleReset": [Function],
          "handleSubmit": [Function],
          "initialErrors": Object {},
          "initialStatus": undefined,
          "initialTouched": Object {},
          "initialValues": Object {
            "email": "",
            "name": "",
            "profile_url": "",
            "user_id": "",
          },
          "isSubmitting": false,
          "isValid": true,
          "isValidating": false,
          "registerField": [Function],
          "resetForm": [Function],
          "setErrors": [Function],
          "setFieldError": [Function],
          "setFieldTouched": [Function],
          "setFieldValue": [Function],
          "setFormikState": [Function],
          "setStatus": [Function],
          "setSubmitting": [Function],
          "setTouched": [Function],
          "setValues": [Function],
          "status": undefined,
          "submitCount": 0,
          "submitForm": [Function],
          "touched": Object {},
          "unregisterField": [Function],
          "validateField": [Function],
          "validateForm": [Function],
          "validateOnBlur": true,
          "validateOnChange": true,
          "validateOnMount": false,
          "values": Object {
            "email": "",
            "name": "",
            "profile_url": "",
            "user_id": "",
          },
        }
      }
      handleClose={[Function]}
      userAlert={
        Object {
          "error": Object {
            "msg": "",
            "status": false,
          },
          "loading": false,
          "success": Object {
            "msg": "",
            "status": false,
          },
        }
      }
    />
  </WithStyles(ForwardRef(Dialog))>
</div>
`;
